e2e/e2etests/test_update_bytecode_connector.go

```diff
@@ -0,0 +1,70 @@
+package e2etests
+
+import (
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/zeta-chain/zetacore/e2e/contracts/testconnectorzevm"
+	"github.com/zeta-chain/zetacore/e2e/runner"
+	"github.com/zeta-chain/zetacore/e2e/utils"
+	fungibletypes "github.com/zeta-chain/zetacore/x/fungible/types"
+)
+
+// TestUpdateBytecodeConnector tests updating the bytecode of a connector and interact with it
+func TestUpdateBytecodeConnector(r *runner.E2ERunner, _ []string) {
+	// Can withdraw 0.1ZETA
+	TestZetaWithdraw(r, []string{"10000000000000000000"})
+
+	// Deploy the test contract
+	newTestConnectorAddr, tx, _, err := testconnectorzevm.DeployTestZetaConnectorZEVM(
+		r.ZEVMAuth,
+		r.ZEVMClient,
+		r.WZetaAddr,
+	)
+	if err != nil {
+		panic(err)
+	}
+
+	// Wait for the contract to be deployed
+	receipt := utils.MustWaitForTxReceipt(r.Ctx, r.ZEVMClient, tx, r.Logger, r.ReceiptTimeout)
+	if receipt.Status != 1 {
+		panic("contract deployment failed")
+	}
+
+	// Get the code hash of the new contract
+	codeHashRes, err := r.FungibleClient.CodeHash(r.Ctx, &fungibletypes.QueryCodeHashRequest{
+		Address: newTestConnectorAddr.String(),
+	})
+	if err != nil {
+		panic(err)
+	}
+	r.Logger.Info("New contract code hash: %s", codeHashRes.CodeHash)
+
+	r.Logger.Info("Updating the bytecode of the Connector")
+	msg := fungibletypes.NewMsgUpdateContractBytecode(
+		r.ZetaTxServer.GetAccountAddress(0),
+		r.ConnectorZEVMAddr.Hex(),
+		codeHashRes.CodeHash,
```

Comment:
Just to understand. The idea is to 1stly write expected byte code to zEVM by `DeployTestZetaConnectorZEVM` and then replace existing connector account's `CodeHash` in order to re-point underlying evm account's code to the newly deployed code. The only useful part of newly deployed contract is its byte code; the contract self (e.g. address) will be just forgotten?

Reply:
>The only useful part of newly deployed contract is its byte code; the contract self (e.g. address) will be just forgotten?

Yes, correct